templates/Default/engine/Default/course_plot.php

```diff
@@ -67,12 +67,19 @@
 						?><option value="<?php echo $Location->getTypeID(); ?>"><?php echo $Location->getName(); ?></option><?php
 					} unset($Location);
 				break;
-				case 'Goods':
-					$Goods =& Globals::getGoods();
+				case 'Sell Goods':
+				case 'Buy Goods':
+					$Goods =& Globals::getVisibleGoods($ThisPlayer);
```

Comment:
It's not huge, but I'd expect this to be:
```php
$Goods =& $ThisPlayer->getVisibleGoods()
```
instead

Reply:
Unfortunately there is no such function. There is one for `SmrPort`, but it only returns those goods that are visible that the port buys/sells.

Reply:
I know, what I meant was that instead of adding it to `Globals` and requiring a player as an arg that having it in SmrPlayer directly would make more sense

Reply:
Oh, I see. I didn't add it to `Globals` though. Would you like me to move the method from `Globals` to `SmrPlayer` as part of this PR?

Reply:
https://github.com/smrealms/smr/pull/190/files#diff-d812d23765b6f995b38f070caf20222bR202 would beg to differ :P

Reply:
LOL! Is my memory that bad? Okay, I'll move it to player :)